apps/wallet-connect/src/hooks/useWalletConnect.tsx

```diff
@@ -56,93 +59,8 @@ const useWalletConnect = () => {
         }
 
         try {
-          let result = '0x'
-
-          switch (payload.method) {
-            case 'eth_sendTransaction': {
-              const txInfo = payload.params[0]
-              const { safeTxHash } = await sdk.txs.send({
-                txs: [
-                  {
-                    to: txInfo.to,
-                    value: txInfo.value || '0x0',
-                    data: txInfo.data || '0x',
-                  },
-                ],
-              })
-
-              result = safeTxHash
-              break
-            }
-            case 'gs_multi_send': {
-              const txs = payload.params
-
-              if (!isMetaTxArray(txs)) {
-                throw new Error('INVALID_TRANSACTIONS_PROVIDED')
-              }
-
-              const { safeTxHash } = await sdk.txs.send({
-                txs: txs.map(txInfo => ({
-                  to: txInfo.to,
-                  value: (txInfo.value || '0x0').toString(),
-                  data: txInfo.data || '0x',
-                })),
-              })
-
-              result = safeTxHash
-              break
-            }
-
-            case 'personal_sign': {
-              const [message, address] = payload.params
-
-              if (!areStringsEqual(address, safe.safeAddress)) {
-                throw new Error('The address or message hash is invalid')
-              }
-
-              await sdk.txs.signMessage(message)
-
-              result = '0x'
-              break
-            }
-
-            case 'eth_sign': {
-              const [address, messageHash] = payload.params
-
-              if (!areStringsEqual(address, safe.safeAddress) || !messageHash.startsWith('0x')) {
-                throw new Error('The address or message hash is invalid')
-              }
-
-              await sdk.txs.signMessage(messageHash)
-
-              result = '0x'
-              break
-            }
-
-            case 'eth_signTypedData':
-            case 'eth_signTypedData_v4': {
-              const [address, typedDataString] = payload.params
-              const typedData = JSON.parse(typedDataString)
-
-              if (!areStringsEqual(address, safe.safeAddress)) {
-                throw new Error('The address is invalid')
-              }
-
-              if (isObjectEIP712TypedData(typedData)) {
```

Comment:
We have this check here but we don't have it on the provider from the SDK. Could it be an issue?

Reply:
The SDK method called by the provider has it: https://github.com/safe-global/safe-apps-sdk/blob/cef74f3ba70f1d5c98e51e83611408faf7982308/packages/safe-apps-sdk/src/txs/index.ts#L49-L51

So I'd say no